app/DTO/ArrayableDTO.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\DTO;
+
+use ReflectionClass;
+use ReflectionProperty;
+
+/**
+ * In some cases, when a DTO does not need to apply casts on attributes
+ * we can directly make use of a reflection which returns an array containing
+ * all the PUBLIC attributes of the DTO.
+ */
+class ArrayableDTO extends DTO
+{
+	/**
+	 * By default, we return an array containing the PUBLIC attributes of the DTO.
+	 *
+	 * @return array the serialized properties of this object
+	 */
+	public function toArray(): array
+	{
+		$result = [];
+		$cls = new ReflectionClass($this);
+		$props = $cls->getProperties(ReflectionProperty::IS_PUBLIC);
+		foreach ($props as $prop) {
+			$result[$prop->getName()] = $prop->getValue($this);
```

Comment:
I wonder if this works in all cases, in particular when the type of the value is not a primitive type, but an object. Maybe this implementation currently works, because all child classes use primitive types only.

But I would have expected something along the following lines for a general approach:

```php
$value = $prop->getValue($this);
if (is_object($value)) {
  if ($value instanceof Arrayable) {
    $value = $value->toArray();
  } elseif (/* some other class type for which we know how to convert */) {
    // convert
  } else {
    throw new \InvalidArgumentException("We don't know how to convert :-(");
  }
$result[$prop->getName()] = $value;
```

Reply:
> Maybe this implementation currently works, because all child classes use primitive types only.

I did not consider other cases, otherwise I would have simplied other DTO using collections for example.